### PR TITLE
Adding format on save to javascript layer

### PR DESCRIPTION
**Pull request by Trapez**
Adding configurable formatting on save functionality similar to typescript
layer. One small syntax error fix in typescript layer.

### CODE REVIEW

File: CHANGELOG.develop [hunk 1]
```diff
@@ -1542,6 +1542,7 @@ Other:
   - Added ES6 module extension to =init-js2-mode= (thanks to Nathan Lloyd)
   - Added =org-babel= support (thanks to Michael Rohleder)
   - Added debug support via =dap= layer (Firefox and Google Chrome)
+  - Added configurable format on save. (Trapez Breen)
```
Comment (robbyoconnor) [inline CHANGELOG.develop]: Can you fix the grammar on this

Comment (Trapez) [reply CHANGELOG.develop]: fixed

**Comment by Trapez** [comment]
@robbyoconnor Updated.

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.

**Pull request merged**